### PR TITLE
Removed futures, added tests, improved readability

### PR DESCRIPTION
**Pull request by flauntingspade4**
As the title suggests, it removes Futures as a dependency, replacing it's only use with a `Vec<Sugar>`. Several other changes, ranging from adding tests for the calculators, improving readibility with changes to whitespace, and implementing a rudimentary temperature conversion system to replace the original

### CODE REVIEW

File: src/calculators/sg_correction.rs [hunk 1]
```diff
@@ -1,28 +1,61 @@
...
+
+pub enum Temperature {
+    Celsius,
+    Farenheight,
+    Kelvin,
+}
+
+pub struct TempParseError;
+
+impl std::convert::TryFrom<char> for Temperature {
```
Comment (drodil) [inline src/calculators/sg_correction.rs:22]: Actually I did this yesterday using the measurements dependency. Benefits of using is that there is also conversions for volumes, weights, etc. (See #25 )

**Review by drodil — COMMENTED** [review]
Looks good but there will be merge conflict with #25 so hoping to get that merged and then this one solved

File: src/calculators/sg_correction.rs (hunk 1 above)
Comment (flauntingspade4) [reply src/calculators/sg_correction.rs:22]: OK, I'll go through and make it compatiable with the measurements dependency

**Comment by mlatief** [comment]
This looks great. Now #25 is merged, please rebase this.

File: src/calculators/sg_correction.rs (hunk 1 above)
Comment (mlatief) [reply src/calculators/sg_correction.rs:22]: This implementation looks better than the initial one in #25 though. Can you modify the now merged units conversion structs after the rebase so they use similar pattern?

File: src/calculators/sg_correction.rs [hunk 2]
```diff
@@ -1,28 +1,61 @@
...
+        let mtf = conversions.to_farenheight(measured_temperature);
 
         return sg * (  (1.00130346 - 0.000134722124 * mtf + 0.00000204052596 * (mtf * mtf) - 0.00000000232820948 * (mtf * mtf * mtf))
                      / (1.00130346 - 0.000134722124 * ctf + 0.00000204052596 * (ctf * ctf) - 0.00000000232820948 * (ctf * ctf * ctf)));
     }
 }
+
+pub enum Temperature {
+    Celsius,
+    Farenheight,
```
Comment (mlatief) [inline src/calculators/sg_correction.rs:16]: *Fahrenheit

**Review by mlatief — CHANGES_REQUESTED** [review]

**Comment by drodil** [comment]
Btw, did you use rustfmt to format these? I think we should enforce the usage with #47 in the wokflow as well.

**Comment by flauntingspade4** [comment]
I did use rustfmt on everything I added, but tried to keep the original formatting the same-I agree, enforcing rustfmt is a good idea to make all the code more readable

File: src/calculators/sg_correction.rs (hunk 1 above)
Comment (drodil) [reply src/calculators/sg_correction.rs:18]: The #25 uses units from the measurements dependency. Only thing done in the conversions.rs is to try and figure out the unit from string. I think you should remove these and use the measurements::Temperature struct instead

Comment (flauntingspade4) [reply src/calculators/sg_correction.rs:18]: So should I just remove what I did in sg_correction.rs, and instead just use the measurements::Temperature struct?

Comment (mlatief) [reply src/calculators/sg_correction.rs:18]: That's one way to do it, also if you feel like changing `conversions.rs` to use `TryFrom` instead, while still using measurements::Temperature please go ahead (can be another PR though)

Comment (flauntingspade4) [reply src/calculators/sg_correction.rs:18]: Ok, I'll change it to be `TryFrom`, then it'll be ready to merge?

Comment (mlatief) [reply src/calculators/sg_correction.rs]: sounds like a plan

**Review by mlatief — APPROVED** [review]
Looks good to me (except the one failing test in `calculators.rs` .. testing  `celsius_to_fahrenheit` .. it can be removed)

File: tests/calculators.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+#[test]
+fn priming() {
+    use rustybeer::calculators::priming::{Priming, Sugar};
+
+    let priming = Priming;
+
+    assert_eq!(59., priming.celsius_to_fahrenheit(15.));
+
+    assert_eq!(53.877197, priming.celsius_to_fahrenheit(12.154));
```
Comment (mlatief) [inline tests/calculators.rs:52]: these two asserts to be removed

File: src/calculators/priming.rs [hunk 4]
```diff
@@ -1,48 +1,53 @@
-use std::error::Error;
-use futures::{future, Future, stream, Stream};
-
 pub struct Priming;
 
 impl Priming {
+    pub fn celsius_to_fahrenheit(&self, celsius: f64) -> f64 {
```
Comment (mlatief) [inline src/calculators/priming.rs:4]: this function isn't needed anymore I guess.

File: src/calculators/sg_correction.rs (hunk 1 above)
Comment (flauntingspade4) [reply src/calculators/sg_correction.rs]: I couldn't see an ergonomic way to change it to `TryFrom`, and have instead have decided to make all the conversion methods return a `Result<Self, ParseFloatError>`, so error handling can be called where the function is called, rather than an immediate `.unwrap()`, upon the parsing failing

File: src/calculators/priming.rs (hunk 4 above)
Comment (flauntingspade4) [reply src/calculators/priming.rs]: I've removed it, and the tests for `celsius_to_fahrenheit`.

**Review by mlatief — CHANGES_REQUESTED** [review]
Some fixes are still needed for it to build and tests to pass

**Comment by flauntingspade4** [comment]
The tests are failing due to the `VolumeBuilder::from_str()` being written incorrectly, I'll have a look into it

File: tests/calculators.rs (hunk 3 above)
Comment (mlatief) [reply tests/calculators.rs:52]: I think these are failing the tests

**Comment by mlatief** [comment]
at least I see the tests failing because of two calls to `celsius_to_fahrenheit` still there

File: src/utils/conversions.rs [hunk 5]
```diff
@@ -32,8 +32,19 @@ impl VolumeBuilder {
...
 
+        match vol.find(':') {
+            Some(count) => {
+                let (temperature, temperature_form) = vol.split_at(count);
+                println!("{}", temperature);
+                println!("{}", temperature_form);
+            }
+            None => {}
+        }
+
```
Comment (mlatief) [inline src/utils/conversions.rs:43]: we don't need these prints

File: src/utils/conversions.rs [hunk 6]
```diff
@@ -32,8 +32,19 @@ impl VolumeBuilder {
...
         if vol.len() > 3 {
             let last3 = &vol[vol.len() - 3..];
+            println!("{}", last3);
+            println!("{}", &vol[..vol.len() - 3]);
```
Comment (mlatief) [inline src/utils/conversions.rs:47]: these too, not needed.

File: src/utils/conversions.rs [hunk 7]
```diff
@@ -1,127 +1,182 @@
...
             };
-
-            if ret.is_ok() {
-                return ret.unwrap();
-            }
-        }
-
-        if vol.len() > 2 {
+            Ok(ret)
+        } else if vol.len() > 2 {
```
Comment (mlatief) [inline src/utils/conversions.rs:59]: I think @drodil had a reason not having them mutually exclusive. One reason comes to mind, if `len` > 2 but the suffix isn't one of `ml`, `m3`, `ul`, or `dr`.

Comment (flauntingspade4) [reply src/utils/conversions.rs]: I've made a different system that's far from perfect, but will parse correctly-the previous one could fail due to lengths of the string while parsing with lower length formats (For example "123ml" would fail, as the length is > 3, so it'd attempt to parse as a three lettered suffix, leading to the return being 12 cubic centimeters, instead of 123 milliliters)

**Comment by flauntingspade4** [comment]
The only remaining conflict is with Cargo.toml where Futures is still listed as a dependency, but I think it makes more sense to get rid of it, and instead just use a `Vec<Sugar>` in the one place it's used, and if a REST service needs to be read from, use `std::io::Read` instead

File: src/utils/conversions.rs (hunk 7 above)
Comment (mlatief) [reply src/utils/conversions.rs]: I think if it can't find `cm3`, it will try again for shorter suffixes, and `ml` will eventually match. The current implementation might not be perfect, but these changes add some unnecessary verbosity to the required inputs, also requires calls to be `unwrap`.

**Review by drodil — APPROVED** [review]
Looks good to me

**Comment by drodil** [comment]
@all-contributors please add @flauntingspade4 for Code
